When running the make under s390x, even having libunwind disabled, I receive the following error
at the build: 
 
make USE_SYSTEM_PCRE=1 PCRE= USE_SYSTEM_*=1 USE_SYSTEM_LIBM=1 USE_SYSTEM_BLAS=1 NO_LAPACK=1 USE_SYSTEM_LLVM=0 DISABLE_LIBUNWIND=1

```
checking for build architecture... s390x
checking for host architecture... s390x
checking for target architecture... s390x
checking for target operating system... linux-gnu
checking for ELF helper width... configure: error: Unknown ELF target: s390x
/data/shared/julia/julia-master/deps/unwind.mk:36: recipe for target 'scratch/libunwind-1.1-julia2/build-configured' failed
make[1]: *** [scratch/libunwind-1.1-julia2/build-configured] Error 1
Makefile:85: recipe for target 'julia-deps' failed
make: *** [julia-deps] Error 2
```

--- a/Make.inc
+++ b/Make.inc
@@ -26,6 +26,7 @@ OPENBLAS_USE_THREAD:=1
 # Use libraries available on the system instead of building them
 USE_SYSTEM_LLVM:=0
 USE_SYSTEM_LIBUNWIND:=0
+DISABLE_LIBUNWIND:=0
 USE_SYSTEM_PCRE:=0
 USE_SYSTEM_LIBM:=0
 USE_SYSTEM_OPENLIBM:=0
@@ -737,6 +738,8 @@ endif
 
 ifeq ($(OS),WINNT)
 LIBUNWIND:=
+else ifneq ($(DISABLE_LIBUNWIND), 0)
+LIBUNWIND:=
 else
 ifeq ($(USE_SYSTEM_LIBUNWIND), 1)
 ifneq ($(OS),Darwin)
@@ -983,6 +986,10 @@ endif
 # OProfile
 ifeq ($(USE_OPROFILE_JITEVENTS), 1)
 JCPPFLAGS += -DJL_USE_OPROFILE_JITEVENTS
+endif
+
+ifeq ($(DISABLE_LIBUNWIND), 1)
+JCPPFLAGS += -DJL_DISABLE_LIBUNWIND
 endif
 
 # Intel libraries

--- a/deps/Makefile
+++ b/deps/Makefile
@@ -48,6 +48,7 @@ ifeq ($(USE_SYSTEM_LIBUV), 0)
 DEP_LIBS += libuv
 endif
 
+ifeq ($(DISABLE_LIBUNWIND), 0)
 ifeq ($(USE_SYSTEM_LIBUNWIND), 0)
 ifeq ($(OS), Linux)
 DEP_LIBS += unwind
@@ -55,6 +56,7 @@ else ifeq ($(OS), FreeBSD)
 DEP_LIBS += unwind
 else ifeq ($(OS), Darwin)
 DEP_LIBS += osxunwind
+endif
 endif
 endif
 

--- a/src/debuginfo.cpp
+++ b/src/debuginfo.cpp
@@ -41,9 +41,7 @@ using namespace llvm;
 #include "julia.h"
 #include "julia_internal.h"
 #include "codegen_internal.h"
-#ifdef _OS_LINUX_
-#  define UNW_LOCAL_ONLY
-#  include <libunwind.h>
+#if defined(_OS_LINUX_)
 #  include <link.h>
 #endif
 
@@ -1400,7 +1398,7 @@ static int jl_getDylibFunctionInfo(jl_frame_t **frames, size_t pointer, int skip
     }
     frame0->fromC = !isSysImg;
     if (isSysImg && sysimg_fvars) {
-#ifdef _OS_LINUX_
+#if defined(_OS_LINUX_) && !defined(JL_DISABLE_LIBUNWIND)
         unw_proc_info_t pip;
         if (!saddr && unw_get_proc_info_by_ip(unw_local_addr_space,
                                               pointer, &pip, NULL) == 0)

--- a/src/julia_internal.h
+++ b/src/julia_internal.h
@@ -648,7 +648,7 @@ typedef struct {
 } bt_cursor_t;
 #endif
 extern volatile int jl_in_stackwalk;
-#else
+#elif !defined(JL_DISABLE_LIBUNWIND)
 // This gives unwind only local unwinding options ==> faster code
 #  define UNW_LOCAL_ONLY
 #  include <libunwind.h>
@@ -660,6 +660,10 @@ typedef unw_cursor_t bt_cursor_t;
 // on a newer release
 #    define JL_UNW_HAS_FORMAT_IP 1
 #  endif
+#else
+// Unwinding is disabled
+typedef int bt_context_t;
+typedef int bt_cursor_t;
 #endif
 size_t rec_backtrace(uintptr_t *data, size_t maxsize);
 size_t rec_backtrace_ctx(uintptr_t *data, size_t maxsize, bt_context_t *ctx);

--- a/src/signals-unix.c
+++ b/src/signals-unix.c
@@ -284,6 +284,7 @@ static void allocate_segv_handler(void)
     }
 }
 
+#if !defined(JL_DISABLE_LIBUNWIND)
 static unw_context_t *volatile signal_context;
 static pthread_mutex_t in_signal_lock;
 static pthread_cond_t exit_signal_cond;
@@ -310,6 +311,7 @@ static void jl_thread_resume(int tid, int sig)
     assert(jl_atomic_load_acquire(&ptls2->signal_request) == 0);
     pthread_mutex_unlock(&in_signal_lock);
 }
+#endif
 
 // Throw jl_interrupt_exception if the master thread is in a signal async region
 // or if SIGINT happens too often.
@@ -360,6 +362,7 @@ void usr2_handler(int sig, siginfo_t *info, void *ctx)
 {
     jl_ptls_t ptls = jl_get_ptls_states();
     sig_atomic_t request = jl_atomic_exchange(&ptls->signal_request, 0);
+#if !defined(JL_DISABLE_LIBUNWIND)
     if (request == 1) {
         signal_context = jl_to_bt_context(ctx);
 
@@ -372,7 +375,9 @@ void usr2_handler(int sig, siginfo_t *info, void *ctx)
         pthread_cond_broadcast(&signal_caught_cond);
         pthread_mutex_unlock(&in_signal_lock);
     }
-    else if (request == 2) {
+    else
+#endif
+    if (request == 2) {
         jl_unblock_signal(sig);
         int force = jl_check_force_sigint();
         if (force || (!ptls->defer_signal && ptls->io_wait)) {
@@ -526,9 +531,7 @@ static void *signal_listener(void *arg)
     static uintptr_t bt_data[JL_MAX_BT_SIZE + 1];
     static size_t bt_size = 0;
     sigset_t sset;
-    unw_context_t *signal_context;
     int sig, critical, profile;
-    int i;
     jl_sigsetset(&sset);
     while (1) {
         profile = 0;
@@ -591,9 +594,11 @@ static void *signal_listener(void *arg)
 #endif
 
         bt_size = 0;
+#if !defined(JL_DISABLE_LIBUNWIND)
+        unw_context_t *signal_context;
         // sample each thread, round-robin style in reverse order
         // (so that thread zero gets notified last)
-        for (i = jl_n_threads; i-- > 0; ) {
+        for (int i = jl_n_threads; i-- > 0; ) {
             // notify thread to stop
             jl_thread_suspend_and_get_state(i, &signal_context);
 
@@ -624,6 +629,7 @@ static void *signal_listener(void *arg)
             // notify thread to resume
             jl_thread_resume(i, sig);
         }
+#endif
 
         // this part is async with the running of the rest of the program
         // and must be thread-safe, but not necessarily signal-handler safe
@@ -647,7 +653,7 @@ void restore_signals(void)
     jl_sigsetset(&sset);
     sigprocmask(SIG_SETMASK, &sset, 0);
 
-#if !defined(HAVE_MACH)
+#if !defined(HAVE_MACH) && !defined(JL_DISABLE_LIBUNWIND)
     if (pthread_mutex_init(&in_signal_lock, NULL) != 0 ||
         pthread_cond_init(&exit_signal_cond, NULL) != 0 ||
         pthread_cond_init(&signal_caught_cond, NULL) != 0) {

--- a/src/stackwalk.c
+++ b/src/stackwalk.c
@@ -13,8 +13,10 @@
 // returning from the callee function will invalidate the context
 #ifdef _OS_WINDOWS_
 #define jl_unw_get(context) RtlCaptureContext(context)
-#else
+#elif !defined(JL_DISABLE_LIBUNWIND)
 #define jl_unw_get(context) unw_getcontext(context)
+#else
+void jl_unw_get(void *context) {};
 #endif
 
 #ifdef __cplusplus
@@ -302,7 +304,7 @@ static int jl_unw_step(bt_cursor_t *cursor, uintptr_t *ip, uintptr_t *sp)
 #endif
 }
 
-#else
+#elif !defined(JL_DISABLE_LIBUNWIND)
 // stacktrace using libunwind
 
 static int jl_unw_init(bt_cursor_t *cursor, bt_context_t *context)
@@ -338,6 +340,18 @@ size_t rec_backtrace_ctx_dwarf(uintptr_t *data, size_t maxsize,
     return n > maxsize ? maxsize : n;
 }
 #endif
+
+#else
+// stacktraces are disabled
+static int jl_unw_init(bt_cursor_t *cursor, bt_context_t *context)
+{
+    return 0;
+}
+
+static int jl_unw_step(bt_cursor_t *cursor, uintptr_t *ip, uintptr_t *sp)
+{
+    return 0;
+}
 #endif
 
 JL_DLLEXPORT jl_value_t *jl_lookup_code_address(void *ip, int skipC)

